blocks/graph.py

```diff
@@ -204,10 +204,15 @@ def replace(self, replacements):
         # Sort `replacements` in topological order
         for node in apply_nodes_sorted:
             for input_ in node.inputs:
-                if input_ not in replacements:
-                    continue
-                replacement_keys_cur.append(input_)
-                replacement_vals_cur.append(replacements[input_])
+                if input_ in replacements:
```

Comment:
Needs a bit of thought. So far it is definitely not because replacements of auxiliary variables is not supported, see #514. If we support it later we will have to be able to traverse all variables including auxiliaries in topological order, but again, the right tool for it would be `self.variables`.

So I would say it's not, maybe we should mention it in a comment in `replace`.